### ISSUE TITLE
LambdaDSL has no implementation for eachLike with an array of primitives

### ISSUE THREAD

**Issue by axulness** [DiUS/pact-jvm#829]
Currently there is no way to define a variable array of x primitive elements via lambda DSL:

```
"componentsIds": [
             "A1",
              ...
             "Ax"
           ]
```
The only alternative is defining a fixed size array:
```
(LambdaDslObject) category.array("componentsIds", modelOptionIdsArray ->{ 
  modelOptionIdsArray.stringType()
  ...
  modelOptionIdsArray.stringType()
});
```
But what would really work here would be an eachLike for an array of primitives.

**Comment by uglyog** [comment]
I've added an `eachLike` method that supports arrays of primitive values.

**Comment by uglyog** [comment]
This has been released with version 3.6.2
